src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -109,7 +108,12 @@ private static object SanitizeValue(object value, string keyName)
         {
             if (value != null)
             {
-                if (value is string || value is bool || value is long || value is double)
+                if (value is string || value is bool || value is double || value is long)
+                {
+                    return value;
+                }
+
+                if (value is string[] || value is bool[] || value is double[] || value is long[])
```

Comment:
won't we support array of int/short/float?

Reply:
done, conversion is made in sanitizevalue